### ISSUE TITLE
Editor upgrade 

### ISSUE THREAD

**Issue by Ineedause** [Anuken/Mindustry-Suggestions#2328]
**Describe the content or mechanics you are proposing.**
Make certain units turn off-able, for example, if you want to have monos in a game, but you don't want flares. If you want to have sei, but no quads. Both of these things are currently impossible to do. 

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**

An update like this will allow for more opportunities to customize a multiplayer map, or even a single player one. Fungal pass without daggers, or pvp maps with monos but no flares. 

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [x] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.
2. - [x] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
3. - [x] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
4. - [x] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
You do know that you need those T1 units to actually make higher tier units right?

**Comment by Kyllingene** [comment]
Monos and flares are in the same tree?

Sei and quad are in the same tree?

**Comment by joshuaptfan** [comment]
Monos and flares are both air units, but they are not in the same tree.

**Comment by Kyllingene** [comment]
> Sei and quad are in the same tree?

I was making the point that this suggestion is really asking for the removal of trees, not individual units in the chain (although, that would be possible by adding one more check before the unit is allowed to exit the reconstructor).

**Comment by itcannotbe** [comment]
> > Sei and quad are in the same tree?
> 
> I was making the point that this suggestion is really asking for the removal of trees, not individual units in the chain (although, that would be possible by adding one more check before the unit is allowed to exit the reconstructor).

Down side is players would get confused why the heck some unit types can't exit factory/reconstructor/payload conveyors.

**Comment by Kyllingene** [comment]
That is true, and I was not advocating that.

**Comment by Dioo0** [comment]
> You do know that you need those T1 units to actually make higher tier units right?
> 

probably he means, that banned units wont be able to move from constructor

**Comment by itcannotbe** [comment]
> > You do know that you need those T1 units to actually make higher tier units right?
> 
> probably he means, that banned units wont be able to move from constructor

https://github.com/Anuken/Mindustry-Suggestions/issues/2328#issuecomment-822890363

**Comment by Ineedause** [comment]
you guys don't understand. It is an example, banning certain units, like in the research tree where you can not produce quads until you have researched them, it is a similar thing. Ban quads, but don't ban monos, ban daggers, but don't ban novas, ban atmbra, but don't ban sei. All of these actions are impossible to do. From a map creator stand point, this will allow for more customization and more challenges, multiplayer or not.

**Comment by genNAowl** [comment]
Will it be possible to see banned units in-game? Because like itcannotbe mentioned, this will be extremely confusing and not even remotely viable otherwise.

**Comment by joshuaptfan** [comment]
If a T1 unit is banned, you can display a red ❌ on the unit in the factory's unit selection pop-up. I don't know how to show a ban for higher tier units, though.

**Comment by itcannotbe** [comment]
> If a T1 unit is banned, you can display a red ❌ on the unit in the factory's unit selection pop-up. I don't know how to show a ban for higher tier units, though.

I think OP's intention is the ability to ban specific units without hampering the ability to produce  the higher tier ones of the same tree.

**Comment by itcannotbe** [comment]
> If a T1 unit is banned, you can display a red ❌ on the unit in the factory's unit selection pop-up. I don't know how to show a ban for higher tier units, though.

But yeah it might be viable to ban a specific tree by just banning the T1 unit from the factory.

**Comment by stuffyAI** [comment]
I don't believe banning the t1 unit of a tree exempts higher tier ones from existence, Units don't exist until they render outside of the factory: I can have the max amount of units on the map and still produce it's upgrades by having the re-constructor adjacent to the respective previous constructor, it just does not leave the factory of the cap is met.

I assume the OP's suggestion can be resolved by force overwriting the normal unit cap, which is usually determined by core level, with 0 for whatever the map meta passes; that way the game just treats the unit cap as always met, but you can still produce its upgrades and we didn't go insane with coding another way of restricting units.

**Comment by itcannotbe** [comment]
> I don't believe banning the t1 unit of a tree exempts higher tier ones from existence, Units don't exist until they render outside of the factory: I can have the max amount of units on the map and still produce it's upgrades by having the re-constructor adjacent to the respective previous constructor, it just does not leave the factory if the cap is met.
> 
> I assume the OP's suggestion can be resolved by force overwriting the normal unit cap, which is usually determined by core level, with 0 for whatever the map meta passes; that way the game just treats the unit cap as always met, but you can still produce its upgrades and we didn't go insane with coding another way of restricting units.

https://github.com/Anuken/Mindustry-Suggestions/issues/2328#issuecomment-822890363

**Comment by stuffyAI** [comment]
> #2328 (comment)

Oh obviously I was expecting the player to get notified of banned units, if we make it hidden then sure make the factory unable to produce the T1s in the first place.
It feels more sensible if you can pick specific units to ban, not only entire groups.
